src/coins.h

```diff
@@ -158,7 +330,7 @@ class CCoinsViewCursor
 class CCoinsView
 {
 public:
-    mutable CSharedCriticalSection cs_utxo;
+    mutable CRecursiveSharedCriticalSection cs_utxo;
 
     //! Retrieve the Coin (unspent transaction output) for a given outpoint.
```

Comment:
not sure I understand why you changed to a recursiveshared lock?  Seems it would be a bit slower to me.